### ISSUE TITLE
NXroot implementation overwrites root attribs given in the template 

### ISSUE THREAD

**Issue by sherjeelshabih** [FAIRmat-NFDI/pynxtools#174]
The current NXroot implementation overwrites root attribs given in the template. In the NXroot attrib write function we should check if the attributes we are writing already exist or not. 

One suggestion is to copy over the attrib value provided in the template. This could be because the user either doesn't know pynxtools already adds them or they want to change the way/one pynxtools is writing currently. The bigger issue is that it is confusing for new users if they think whatever data we provide in the Template should go in the NeXus file.

The other suggestion is to give an error if the attrib provided by the user is different in value then what the default write function provides. This also makes sense for sensitive attributes like Application Definition or NeXus Version. But if the user really wants to override, we will need to provide a flag for that.

We need a decision on what suits better.

This is related to #163.

**Comment by sherjeelshabih** [comment]
Leaving this here for the future:

>Yes I would vote for this option from #174:

>> The other suggestion is to give an error if the attrib provided by the user is different in value then what the default write function provides. This also makes sense for sensitive attributes like Application Definition or NeXus Version. But if the user really wants to override, we will need to provide a flag for that.

>And I think we should also not just give a flag but error and explain to the user how they may achieve what they want (e.g., changing the definitions to another submodule and using another set of appdefs or use another version of pynxtools). Because I honestly don't see a reason why we should allow users to just change this value from which we know that we use another one for creating the file.

_Originally posted by @domna in https://github.com/FAIRmat-NFDI/pynxtools/pull/124#discussion_r1378462647_

**Comment by RubelMozumder** [comment]
We close this issue. The original issue #124 is already merged, and the root attribute we are adding in this [function](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/helpers.py#L1166) should not be polluted. So, if users try to give any value corresponding to those proprietary attributes, we can overwrite the user-provided values. For such overwritten changes, the function is throwing a warning message. @sherjeelshabih, what do you say?

**Comment by lukaspie** [comment]
> We close this issue. The original issue [#124](https://github.com/FAIRmat-NFDI/pynxtools/pull/124) is already merged, and the root attribute we are adding in this [function](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/helpers.py?rgh-link-date=2025-10-01T14%3A41%3A47.000Z#L1166) should not be polluted. So, if users try to give any value corresponding to those proprietary attributes, we can overwrite the user-provided values. For such overwritten changes, the function is throwing a warning message. [@sherjeelshabih](https://github.com/sherjeelshabih) and [@lukaspie](https://github.com/lukaspie) what do you say?

Exactly, given the presense of the [`update_and_warn_function`](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/helpers.py#L1171), this is a solved issue. Thanks for checking.
